### ISSUE TITLE
[BUG] `ConformalIntervals` `conformal_bonferroni` method doesn't work as expected when alpha = 0.5

### ISSUE THREAD

**Issue by bethrice44** [sktime/sktime#3693]
**Describe the bug**
When using `predict_quantiles` with `ConformalIntervals`, you would expect `alpha=0.5` to correspond to the output of `predict` (or very close to it). However, when the Bonferroni correction is applied, it doesn't at all and the error becomes large vs just using `predict`.

I think this is due to the way `_predict_quantiles` is converted to `_predict_interval` and how zero coverage is interpreted in the logic.

**To Reproduce**
Toy example using the logic from `conformal_bonferroni` correction [code](https://github.com/sktime/sktime/blob/ad2619ba76675c9b8a9bc9174e5e442ca4730284/sktime/forecasting/conformal.py#L243):

```python
# Set alpha to 0.5 and a toy forecasting horizon and abs_residuals
alpha = 0.5
fh = [1,2,3,4,5]
abs_resids = [0.1, 0.2, 0.3, 0.25, 0.1, 0.0001]

# alpha is converted to coverage in `_predict_quantiles` base forecaster 
coverage = np.abs(1 - 2 * alpha)
# coverage = 0 

coverage2 = np.repeat(coverage, 2)
# coverage2 = [0, 0]

alphas = 1 - coverage2
# alphas = [1, 1]

quantiles = 1 - alphas / len(fh)
# quantiles = [0.8, 0.8]

pred_int_row = np.quantile(abs_resids, quantiles)
# pred_int_row = [0.3, 0.3]

I think even in the `conformal` logic this is (less) incorrect because you take the 0th quantile which is the smallest residual and I'm not sure if this is what you want to do. 

In the toy example:
```
quantiles = coverage2
pred_int_row = np.quantile(abs_resids, quantiles)
# pred_int_row = [0.0001, 0.0001]
```

```

**Expected behavior**
`_predict_interval` with alpha = 0.5 and therefore coverage = 0 should return `y_pred + 0`

**Versions**
0.13.4
System:
    python: 3.8.13 (default, May  8 2022, 17:48:02)  [Clang 13.1.6 (clang-1316.0.21.2)]
executable: /Users/bethrice/Documents/Code/uk-ds-day-ahead-prices-forecast/.venv/bin/python
   machine: macOS-12.1-arm64-i386-64bit

Python dependencies:
          pip: 22.1.2
   setuptools: 63.1.0
      sklearn: 1.1.2
       sktime: 0.13.4
  statsmodels: 0.13.2
        numpy: 1.22.0
        scipy: 1.8.1
       pandas: 1.4.1
   matplotlib: 3.6.0
       joblib: 1.2.0
        numba: 0.56.2
     pmdarima: None
      tsfresh: None

</details>

**Comment by ltsaprounis** [comment]
> When using `predict_quantiles` with `ConformalIntervals`, you would expect `alpha=0.5` to correspond to the output of `predict` (or very close to it)

`alpha=0.5` should return the median right?

**Comment by fkiraly** [comment]
>So a difference between these is expected and for some time series the difference will be significant.

Yes, that is true, this might explain the issue, but I feel it does not explain it completely (see the calculation that @bethrice44 did).
